### DISCUSSION TITLE
How to customize/extend MACE models

### DISCUSSION THREAD

**Discussion by peastman** [ACEsuit/mace#984] [Ideas]
Is it possible to customize a MACE model beyond the standard fixed options?  Apologies if this is documented somewhere.  I searched the documentation but didn't find anything.  Here are some particular things I would like to do.

- Add extra readouts for additional atomic or molecular properties I want to predict.  These could be things like partial charge, electronegativity, solubility, etc.  Different outputs may require different nonlinearities.
- Add another term to the loss function based on how well the extra outputs match values stored in the training set.
- Add an extra term to the energy that is computed with code I write myself, for example to implement a physics based energy term.
- Add more atomic properties that are incorporated into the initial embedding.  That is, I want to describe each atom by more than just a one-hot encoded element index.

Are any of these things possible?

**Comment by gabor1** [comment]
hi Peter!

Let's discuss your points in more detail. All these things are possible, but obviously require coding. Note that specifically for electrostatics several electrostatic models are in the works. There is an active PR for implementing Bingqing Cheng's LES model, it's basically ready to try if you clone the version from the PR. We are also working on an SCF version that will be ready soon. 

Happy to have a call on your points of interest. 

Gabor

**Comment by peastman** [comment]
A call would be great.  I'll email you.
